src/coreComponents/fileIO/timeHistory/HistoryCollectionBase.hpp

```diff
@@ -0,0 +1,123 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2020-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+#ifndef GEOSX_HISTORYCOLLECTIONBASE_HPP
+#define GEOSX_HISTORYCOLLECTIONBASE_HPP
+
+#include "HistoryCollection.hpp"
+
+#if defined(GEOSX_USE_PYGEOSX)
+#include "fileIO/python/PyHistoryCollectionType.hpp"
+#endif
+
+namespace geosx
+{
+
+/**
+ * @brief Technical code factorisation after refactoring. Feel free to challenge this class.
```

Comment:
I'm fine with this class, but we need to document or remove this comment

Reply:
Is `Intermediate class for code factorisation` OK for you? Because this is what I wanted to express.

Reply:
I'd go slightly stronger with `Intermediate class solely for code factorisation`, but honestly either is fine.

Reply:
This is fine I'll go that way.